esx_service/utils/auth.py

```diff
@@ -0,0 +1,270 @@
+import auth_data_const
+
+# All supported vmdk commands
+CREATE = 'create'
+REMOVE = 'remove'
+LIST = 'list'
+GET = 'get'
+ATTACH = 'attach'
+DETACH = 'detach'
+
+SIZE = 'size'
+
+_auth_mgr = None
+def connect_auth_db():
+    """ Get a connection to auth DB. """
+    global _auth_mgr
+    if not _auth_mgr:
+        _auth_mgr = auth_data.AuthorizationDataManager()
+        _auth_mgr.connect()
+       
+def get_tenant(vm_uuid):
+    """ Get tenant which owns this VM by querying the auth DB. """
+    global _auth_mgr
+    try:
+        cur = _auth_mgr.conn.execute(
+                    "SELECT tenant_id FROM vms WHERE vm_id = ?",
+                    (vm_uuid, )
+        )
+        result = cur.fetchone()
+    except sqlite3.Error, e:
+        logging.error("Error %s when querying from vms table for vm_id %s", e, vm_uuid)
+    if result:
+        logging.debug("get tenant vm_uuid=%s tenant_id=%s", vm_uuid, result[0])
+   
+    tenant_uuid = None
+    tenant_name = None
+    if result:
+        tenant_uuid = result[0]
+        try:
+            cur = _auth_mgr.conn.execute(
+                    "SELECT name FROM tenants WHERE id = ?",
+                    (tenant_uuid, )
+            )
+            result = cur.fetchone()
+        except sqlite3.Error, e:
+            logging.error("Error %s when querying from tenants table for tenant_id %s", e, tenant_uuid)
+
+        if result:
+            tenant_name = result[0]
+            logging.debug("Found tenant_uuid %s, tenant_name", tenant_uuid, tenant_name)
+
+    return tenant_uuid, tenant_name
+
+def get_privileges(tenant_uuid, datastore):
+    """ Return privileges for given (tenant_uuid, datastore) pair by
+        querying the auth DB.
+
+    """
+    global _auth_mgr
+    privileges = []
+    logging.debug("get_privileges tenant_uuid=%s datastore=%s", tenant_uuid, datastore)
+    try:
+        cur = _auth_mgr.conn.execute(
+                    "SELECT * FROM privileges WHERE tenant_id = ? and datastore = ?",
+                    (tenant_uuid, datastore)    
+        )
+        privileges = cur.fetchone()
+    except sqlite3.Error, e:
+        logging.error("Error %s when querying privileges table for tenant_id %s and datastore %s", 
+                      e, tenant_uuid, datastore)
+    return privileges
+
+def has_privilege(privileges, type):
+    """ Check the privileges has the specific type of privilege set. """
+    logging.debug(privileges)
+    if not privileges:
+        return False
+    logging.debug("%s=%d", type, privileges[type])
+    return privileges[type]
+
+def get_vol_size(opts):
+    """ get volume size. """
+    if not opts or not opts.has_key(SIZE):
+        logging.error("Volume size not specified")
+    return opts[SIZE]    
+
+
+def check_max_volume_size(opts, privileges):
+    """ Check wheter the size of the volume to be created exceeds
+        the max volume size specified in the privileges.
+
+    """
+    vol_size_in_MB = convert.convert_to_MB(get_vol_size(opts))
```

Comment:
Done. Change get_vol_size to return default value is 'size" is not specified in opts.